File: crates/compiler/zo-benches/benches/fibonacci/fibonacci.c
#include <assert.h>
#include <stdio.h>

int fib(int n) {
  if (n == 0) {
    return 0;
  } else {
    if (n <= 2) {
      return 1;
    } else {
      return fib(n - 1) + fib(n - 2);
    }
  }
}

int main() {
  assert(fib(8) == 21);
  assert(fib(15) == 610);
  printf("%d\n", fib(8));
  printf("%d\n", fib(15));

  return 0;
}

File: crates/compiler/zo-benches/benches/fibonacci/fibonacci.rs
fn fib(n: i64) -> i64 {
  if n == 0 {
    return 0;
  } else {
    if n <= 2 {
      return 1;
    } else {
      return fib(n - 1) + fib(n - 2);
    }
  }
}

fn main() {
  assert_eq!(fib(8), 21);
  assert_eq!(fib(15), 610);
  println!("{}", fib(8));
  println!("{}", fib(15));
}
